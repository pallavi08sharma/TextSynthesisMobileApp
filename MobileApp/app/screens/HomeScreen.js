import React from "react";
import { ImageBackground, StyleSheet, View, Image, Text } from "react-native";

import AppButton from "../components/AppButton";
import colors from "../config/colors";
import routes from "../navigation/routes";

function HomeScreen({ navigation }) {
  return (
    <ImageBackground
      blurRadius={0.8}
      style={styles.background}
      source={require("../assets/background.jpeg")}
    >
      <View style={styles.logoContainer}>
        <Image style={styles.logo} source={require("../assets/logoHome.png")} />
        <Text style={styles.tagline}>Synthesize Long Texts</Text>
      </View>
      <View style={styles.buttonsContainer}>
        <AppButton
          title="Login"
          onPress={() => navigation.navigate(routes.LOGIN)}
        />
        <AppButton
          title="Register"
          color="secondary"
          onPress={() => navigation.navigate(routes.REGISTER)}
        />
      </View>
    </ImageBackground>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    justifyContent: "flex-end",
    alignItems: "center",
  },
  buttonsContainer: {
    padding: 20,
    width: "100%",
  },
  logo: {
    width: 300,
    height: 180,
  },
  logoContainer: {
    position: "absolute",
    top: 1,
    alignItems: "center",
  },
  tagline: {
    fontSize: 16,
    fontWeight: "bold",
    paddingVertical: 1,
    color: colors.primary,
  },
});

export default HomeScreen;