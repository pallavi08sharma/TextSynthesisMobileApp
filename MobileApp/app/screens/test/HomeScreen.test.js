import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from "../HomeScreen"


const navigate=jest.fn();

describe('<HomeScreen />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree.children.length).toBe(3);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});