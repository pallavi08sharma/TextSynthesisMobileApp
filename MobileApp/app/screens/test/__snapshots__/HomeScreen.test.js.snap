// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <Image
    blurRadius={0.8}
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "position": "absolute",
        "top": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 180,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "deepskyblue",
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingVertical": 1,
        }
      }
    >
      Synthesize Long Texts
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "deepskyblue",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 15,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "grey",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 15,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
</View>
`;
